File: run-migration.mjs
import { spawn } from 'child_process';

const child = spawn('npx', ['drizzle-kit', 'generate', '--name', 'rev71'], {
  cwd: '/home/ubuntu/erp-rh-fc',
  stdio: ['pipe', 'pipe', 'pipe'],
});

let output = '';

child.stdout.on('data', (data) => {
  const text = data.toString();
  output += text;
  process.stdout.write(text);
  
  // When we see a column conflict prompt, press Enter to select "create column" (default)
  if (text.includes('create column') || text.includes('rename column')) {
    setTimeout(() => {
      child.stdin.write('\n');
    }, 100);
  }
});

child.stderr.on('data', (data) => {
  process.stderr.write(data.toString());
});

child.on('close', (code) => {
  console.log(`\n--- Process exited with code ${code} ---`);
  process.exit(code);
});

// Safety timeout
setTimeout(() => {
  console.log('\n--- Timeout, killing process ---');
  child.kill();
  process.exit(1);
}, 120000);

File: run-migration.py
#!/usr/bin/env python3
import pexpect
import sys

child = pexpect.spawn('npx drizzle-kit generate --name rev71', cwd='/home/ubuntu/erp-rh-fc', timeout=120)
child.logfile_read = sys.stdout.buffer

while True:
    try:
        idx = child.expect([
            r'create column',
            r'1 migration',
            r'migrations generated',
            r'No schema changes',
            pexpect.EOF,
            pexpect.TIMEOUT
        ], timeout=30)
        
        if idx == 0:
            # Press Enter to select "create column" (default option)
            child.sendline('')
        elif idx in [1, 2, 3]:
            # Migration generated or no changes
            print("\n\n=== Migration complete ===")
            break
        elif idx == 4:
            print("\n\n=== EOF ===")
            break
        elif idx == 5:
            print("\n\n=== TIMEOUT ===")
            break
    except Exception as e:
        print(f"\n\n=== Error: {e} ===")
        break

child.close()
print(f"\nExit code: {child.exitstatus}")
